docker/Makefile

```diff
@@ -0,0 +1,54 @@
+git_describe := $(shell git describe)
+vcs_ref := $(shell git rev-parse HEAD)
+build_date := $(shell date -u +%FT%T)
+hadolint_available := $(shell hadolint --help > /dev/null 2>&1; echo $$?)
+hadolint_command := hadolint --ignore DL3008 --ignore DL3018 --ignore DL4000 --ignore DL4001
+hadolint_container := hadolint/hadolint:latest
+ifeq ($(IS_NIGHTLY),true)
+	dockerfile := Dockerfile.nightly
+	version := puppet6-nightly
+else
+	version := $(shell echo $(git_describe) | sed 's/-.*//')
+	dockerfile := Dockerfile
+endif
+
+
+prep:
+ifeq ($(IS_NIGHTLY),true)
+	@echo "No prep needed!"
+else
+	@git pull --unshallow > /dev/null 2>&1 ||:
+	@git fetch origin 'refs/tags/*:refs/tags/*' > /dev/null 2>&1
+endif
+
+lint:
+ifeq ($(hadolint_available),0)
+	@$(hadolint_command) puppetserver-standalone/$(dockerfile)
+	@$(hadolint_command) puppetserver/$(dockerfile)
+else
+	@docker pull $(hadolint_container)
+	@docker run --rm -v $(PWD)/puppetserver-standalone/$(dockerfile):/Dockerfile -i $(hadolint_container) $(hadolint_command) Dockerfile
+	@docker run --rm -v $(PWD)/puppetserver/$(dockerfile):/Dockerfile -i $(hadolint_container) $(hadolint_command) Dockerfile
+endif
+
+build: prep
+	@docker build --build-arg vcs_ref=$(vcs_ref) --build-arg build_date=$(build_date) --build-arg version=$(version) --file puppetserver-standalone/$(dockerfile) --tag puppet/puppetserver-standalone:$(version) puppetserver-standalone
+	@docker build --build-arg vcs_ref=$(vcs_ref) --build-arg build_date=$(build_date) --build-arg version=$(version) --file puppetserver/$(dockerfile) --tag puppet/puppetserver:$(version) puppetserver
+ifeq ($(IS_LATEST),true)
+	@docker tag puppet/puppetserver-standalone:$(version) puppet/puppetserver-standalone:latest
+	@docker tag puppet/puppetserver:$(version) puppet/puppetserver:latest
+endif
+
+test: prep
+	@PUPPET_TEST_DOCKER_IMAGE=puppet/puppetserver-standalone:$(version) rspec puppetserver-standalone/spec
+	@PUPPET_TEST_DOCKER_IMAGE=puppet/puppetserver:$(version) rspec puppetserver/spec
+
+publish: prep
+	@docker push puppet/puppetserver-standalone:$(version)
+	@docker push puppet/puppetserver:$(version)
+ifeq ($(IS_LATEST),true)
```

Comment:
Where does this `IS_LATEST` come from? 

Hadn't noticed this before, thought there was only `IS_NIGHTLY`.

Reply:
IS_LATEST is an environment variable that we pass in during the pipelines job. Initially we were needing to edit jobs to get the latest tag built for the correct branch and that was getting really weird with mergeups

Reply:
And set it to `$(version)` if not already set to `latest`

```make
container_version ?= $(version)

@docker push puppet/puppetserver-standalone:$(container_version)
@docker push puppet/puppetserver:$(container_version)
```